src/mapWithFeedback.ts

```diff
@@ -0,0 +1,130 @@
+import { _reduceLazy } from "./_reduceLazy";
+import { _toLazyIndexed } from "./_toLazyIndexed";
+import type { LazyEvaluator } from "./pipe";
+import { purry } from "./purry";
+
+/**
+ * Applies a reducer function on each element of the array, accumulating the results, and returns an array of the successively reduced values.
```

Comment:
The term accumulator is not accurate here. The function could be used to accumulate results, but it could also be used to compute any other form of computation with memory, like sum.

Even the typescript library itself doesn't call the accumulator value in `reduce` accumulator, opting for the more accurate `previousValue`.

https://github.com/microsoft/TypeScript/blob/main/src/lib/es5.d.ts#L1463-L1470

Reply:
I agree. I think it's best to keep the parameter names consistent with the typescript team's conventions as well.
I'm opting to also change the reducer parameter name to callbackFn even though it's less descriptive, so that people can "pattern match" the two signatures and infer that the usage will be similar without learning any new API pattern.

If this sounds good, I'll introduce that into the PR as well.

Reply:
Added requested changes.

Reply:
I totally agree about `callbackfn`, notice that it's all lowercase!